src/main/java/tech/ydb/spark/connector/YdbTableProvider.java

```diff
@@ -85,8 +87,16 @@ private YdbTable getOrLoadTable(Map<String, String> props) {
             dts.setIncludeShardKeyBounds(ti.indexName == null); // shard keys for table case
             Result<TableDescription> tdRes = session.describeTable(ti.tablePath, dts).join();
             if (!tdRes.isSuccess()) {
-                LOG.debug("Failed to load table description for {}: {}", ti.tablePath, tdRes.getStatus());
-                return CompletableFuture.completedFuture(Result.fail(tdRes.getStatus()));
+                if (schema == null) {
+                    LOG.debug("Failed to load table description for {}: {}", ti.tablePath, tdRes.getStatus());
+                    return CompletableFuture.completedFuture(Result.fail(tdRes.getStatus()));
+                }
+                // not such table => create a new, according the schema of given dataframe
+                YdbCatalog catalog = new YdbCatalog();
```

Comment:
I am not 100% sure, but there should be a better place for implicit table creation. `df.save.format("ydb")` works already without that addition.
Probably an example of Spark code requiring such change will help to understand why was that requested.

Reply:
> I am not 100% sure, but there should be a better place for implicit table creation

think so too. Felt really out of place, especially building catalog obj in this context

----
> `df.save.format("ydb")` works already without that addition.
Probably an example of Spark code requiring such change will help to understand why was that requested.

not quite. It fails when there's no such table yet. It's not quite typical behaviour for spark job iirc. As an example of the code when it fails you may use `tech.ydb.spark.connector.integration.IntegrationTest#testViaJavaApi` testcase with YdbTableProvider unmodified

Reply:
about this class' logic in general:

in other connectors `TableProvider` implementations are usually pretty thin and just build `Table` objects based on provided schema. Schema introspection mechanism usually lives elsewhere.

Compared with other connectors using this approach: Mongo, Couchbase (columnar), Dynamo